glean-core/rlb/tests/schema.rs

```diff
@@ -48,39 +52,63 @@ fn new_glean() -> tempfile::TempDir {
 fn validate_against_schema() {
     let schema = load_schema();
 
-    let dir = new_glean();
-
-    // Register and submit a ping for testing
-    let ping_type = PingType::new("test", true, /* send_if_empty */ true, vec![]);
-    glean::register_ping_type(&ping_type);
-    ping_type.submit(None);
-    glean::dispatcher::block_on_queue();
-
-    // Read the ping from disk.
-    // We know where it should be placed.
-    let pings_dir = dir.path().join("pending_pings");
-    let pending_pings: Vec<PathBuf> = pings_dir
-        .read_dir()
-        .unwrap()
-        .filter_map(|entry| entry.ok())
-        .filter_map(|entry| match entry.file_type() {
-            Ok(f) if f.is_file() => Some(entry.path()),
-            _ => None,
-        })
-        .collect();
-
-    // There should only be one: our custom ping.
-    assert_eq!(1, pending_pings.len());
-
-    // 1. line: the URL
-    // 2. line: the JSON body
-    let file = File::open(&pending_pings[0]).unwrap();
-    let mut lines = BufReader::new(file).lines();
-    let _url = lines.next().unwrap().unwrap();
-    let body = lines.next().unwrap().unwrap();
+    let (s, r) = crossbeam_channel::bounded::<Vec<u8>>(1);
+
+    // Define a fake uploader that reports back the submission URL
```

Comment:
It reports back the body, not the URL

```suggestion
    // Define a fake uploader that reports back the submitted payload
```